packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -184,6 +251,57 @@ export const makeWalletBridgeFromFollower = (
   );
   const { acceptOffer, declineOffer, cancelOffer } = offerService;
 
+  // We override addOffer to adapt the old proposalTemplate format to the new
+  // smart-wallet format.
+  const addOfferPSMHack = async details => {
+    const {
+      instanceHandleBoardId: instance, // This actually is the instance handle, not an ID.
+      invitationMaker: { method },
+      proposalTemplate: { give, want },
+    } = details;
+
+    const mapPurses = obj =>
+      Object.fromEntries(
+        Object.entries(obj).map(([kw, { brand, pursePetname, value }]) => [
+          kw,
+          {
+            brand: brand || pursePetnameToBrand.get(pursePetname),
+            value: BigInt(value),
+          },
+        ]),
+      );
+    const offer = {
+      id: new Date().getTime(),
+      invitationSpec: {
```

Comment:
This is what I was thinking of earlier, the dapp should probably be sending `invitationSpec` as-is instead of having to pull out `method` and `instance` from the details separately. Only the AMM and PSM use those, they were added just for smart wallet.

Reply:
Let's consider that as a future API cleanup, but I'm not confident in executing that without making deeper changces to the dapp UI logic, with which I'm not really familiar.